src/main/java/org/junitpioneer/jupiter/RangeSourceProvider.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.support.AnnotationConsumer;
+import org.junit.platform.commons.util.Preconditions;
+
+/**
+ * Provides a range of {@code int}s, as defined by {@link RangeSource}.
+ *
+ * @see RangeSource
+ */
+public class RangeSourceProvider implements ArgumentsProvider, AnnotationConsumer<RangeSource> {
+	/**
+	 * The beginning of the range, inclusive.
+	 */
+	private int[] values;
```

Comment:
[Extensions have to be stateless](https://blog.codefx.org/design/architecture/junit-5-extension-model/#Stateless), so we can't use a field here. I recommend to compute the values in `provideArguments`.

Reply:
Actually, in contrast to an `Extension`, an `ArgumentsProvider` that implements `AnnotationConsumer` is (always?) stateful. 😉

Reply:
Oh, have to catch up on that. :blush: @mureinik: Nevermind, then.